bundles/org.openhab.core/src/main/java/org/openhab/core/items/GenericItem.java

```diff
@@ -251,18 +251,23 @@ protected void notifyListeners(final State oldState, final State newState) {
         Set<StateChangeListener> clonedListeners = null;
         clonedListeners = new CopyOnWriteArraySet<>(listeners);
         ExecutorService pool = ThreadPoolManager.getPool(ITEM_THREADPOOLNAME);
-        for (final StateChangeListener listener : clonedListeners) {
-            pool.execute(() -> {
-                try {
-                    listener.stateUpdated(GenericItem.this, newState);
-                    if (newState != null && !newState.equals(oldState)) {
-                        listener.stateChanged(GenericItem.this, oldState, newState);
+        try {
+            final boolean stateChanged = newState != null && !newState.equals(oldState);
+            clonedListeners.forEach(listener ->
+                pool.execute(() -> {
+                    try {
+                        listener.stateUpdated(GenericItem.this, newState);
+                        if (stateChanged) {
+                            listener.stateChanged(GenericItem.this, oldState, newState);
+                        }
+                    } catch (Exception e) {
+                        logger.warn("failed notifying listener '{}' about state update of item {}: {}", listener,
+                                GenericItem.this.getName(), e.getMessage(), e);
                     }
-                } catch (Exception e) {
-                    logger.warn("failed notifying listener '{}' about state update of item {}: {}", listener,
-                            GenericItem.this.getName(), e.getMessage(), e);
-                }
-            });
+                }));
+        } catch (Exception e) {
+            logger.warn("failed comparing oldState '{}' to newState '{}' about state update of item {}: {}", oldState,
```

Comment:
Minor point - I assume that this should say -:

```
failed comparing oldState '{}' to newState '{}' during state update of item
```

or

```
failed comparing oldState '{}' to newState '{}' while updating state of item
```

or something similar - basically the "about" doesn't seem right.

(I told you it was minor - sorry for being pick :) ).

Reply:
Copy&Paste error... Will fix it.